sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala

```diff
@@ -599,6 +602,37 @@ abstract class BaseScriptTransformationSuite extends SparkPlanTest with SQLTestU
           'e.cast("string")).collect())
     }
   }
+
+  test("SPARK-35220: DayTimeIntervalType/YearMonthIntervalType show different " +
+    "between hive serde and row format delimited\t") {
+    assume(TestUtils.testCommandAvailable("/bin/bash"))
+    withTempView("v") {
+      val df = Seq(
+        (Duration.ofDays(1), Period.ofMonths(10))
+      ).toDF("a", "b")
+      df.createTempView("v")
+
+      if (defaultSerDe == "hive-serde") {
+        checkAnswer(sql(
+          """
+            |SELECT TRANSFORM(a, b)
+            |  USING 'cat' AS (a, b)
+            |FROM v
+            |""".stripMargin),
+          identity,
+          Row("1 00:00:00.000000000", "0-10") :: Nil)
+      } else {
+        checkAnswer(sql(
+          """
+            |SELECT TRANSFORM(a, b)
+            |  USING 'cat' AS (a, b)
+            |FROM v
+            |""".stripMargin),
+          identity,
+          Row("INTERVAL '1 00:00:00' DAY TO SECOND", "INTERVAL '0-10' YEAR TO MONTH") :: Nil)
```

Comment:
so the spark-sql shell and `df.show` have different formats for intervals?

Reply:
> so the spark-sql shell and `df.show` have different formats for intervals?

Yea, have this problem too, since spark sql follow hive format. What should I to do next?

Reply:
Is interval format the only difference between hive format and spark cast?

Reply:
> Is interval format the only difference between hive format and spark cast?

Yea， ANSI_STYLE and HIVE_STYLE

Reply:
Maybe we should have a new Expression `ToHiveString` and use it in `df.show`, so that they are consistent.

Reply:
> Maybe we should have a new Expression `ToHiveString` and use it in `df.show` and `TRANSFORM`, so that they are consistent.

Yea,  create a ticket https://issues.apache.org/jira/browse/SPARK-35228